service/awsconfig/v8/resource/cloudformation/delete.go

```diff
@@ -11,72 +11,96 @@ import (
 	}
-	r.logger.LogCtx(ctx, "level", "debug", "deleting AWS Host Pre-Guest CloudFormation stack: deleted")
 
-	deleteStackInput = awscloudformation.DeleteStackInput{
-		StackName: aws.String(key.MainHostPostStackName(cluster)),
+	if stackInputToDelete.Name != "" {
+		r.logger.LogCtx(ctx, "level", "debug", "message", "deleting the host cluster pre stack")
+
+		i := &awscloudformation.DeleteStackInput{
+			StackName: aws.String(key.MainHostPreStackName(customObject)),
+		}
+		_, err = r.hostClients.CloudFormation.DeleteStack(i)
+		if err != nil {
+			return microerror.Mask(err)
+		}
+
+		r.logger.LogCtx(ctx, "level", "debug", "message", "deleted the host cluster pre stack")
+	} else {
+		r.logger.LogCtx(ctx, "level", "debug", "message", "not deleting the host cluster pre stack")
 	}
-	_, err = r.hostClients.CloudFormation.DeleteStack(&deleteStackInput)
-	if err != nil {
-		return microerror.Maskf(err, "deleting AWS Host Post-Guest CloudFormation Stack")
+
+	if stackInputToDelete.Name != "" {
+		r.logger.LogCtx(ctx, "level", "debug", "message", "deleting the host cluster post stack")
+
+		i := &awscloudformation.DeleteStackInput{
+			StackName: aws.String(key.MainHostPostStackName(customObject)),
+		}
+		_, err = r.hostClients.CloudFormation.DeleteStack(i)
+		if err != nil {
+			return microerror.Mask(err)
+		}
+
+		r.logger.LogCtx(ctx, "level", "debug", "message", "deleted the host cluster post stack")
+	} else {
+		r.logger.LogCtx(ctx, "level", "debug", "message", "not deleting the host cluster post stack")
 	}
-	// TODO align with integration tests
-	r.logger.LogCtx(ctx, "level", "debug", "deleting AWS Host Post-Guest CloudFormation stack: deleted")
 
 	return nil
 }
 
 func (r *Resource) NewDeletePatch(ctx context.Context, obj, currentState, desiredState interface{}) (*framework.Patch, error) {
-	delete, err := r.newDeleteChange(ctx, obj, currentState, desiredState)
+	deleteChange, err := r.newDeleteChange(ctx, obj, currentState, desiredState)
 	if err != nil {
 		return nil, microerror.Mask(err)
 	}
 
 	patch := framework.NewPatch()
-	patch.SetDeleteChange(delete)
+	patch.SetDeleteChange(deleteChange)
 
 	return patch, nil
 }
 
 func (r *Resource) newDeleteChange(ctx context.Context, obj, currentState, desiredState interface{}) (interface{}, error) {
-	currentStackState, err := toStackState(currentState)
-	if err != nil {
-		return awscloudformation.DeleteStackInput{}, microerror.Mask(err)
-	}
 	desiredStackState, err := toStackState(desiredState)
 	if err != nil {
-		return awscloudformation.DeleteStackInput{}, microerror.Mask(err)
+		return StackState{}, microerror.Mask(err)
 	}
 
-	r.logger.LogCtx(ctx, "level", "debug", "finding out if the main stack should be deleted")
+	r.logger.LogCtx(ctx, "level", "debug", "message", "finding out if the guest cluster main stack has to be deleted")
 
-	deleteState := awscloudformation.DeleteStackInput{
-		StackName: aws.String(""),
-	}
+	var deleteState StackState
+
+	if desiredStackState.Name != "" {
```

Comment:
We are actually only using the desired state now. I couldn't find a way in which the current state is useful and there also might be situations in which we maybe not be able to compute the current state but want to delete anyway. WDYT?

Reply:
Makes sense. We just need to detect that the stacks should be deleted.

Reply:
Yes, current state is not useful here.